server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -1576,12 +1576,12 @@ protected Metadata buildMetadataWithPreviousIndicesLookups() {
                 indices,
                 templates,
                 customs,
-                Arrays.copyOf(previousMetadata.allIndices, previousMetadata.allIndices.length),
-                Arrays.copyOf(previousMetadata.visibleIndices, previousMetadata.visibleIndices.length),
-                Arrays.copyOf(previousMetadata.allOpenIndices, previousMetadata.allOpenIndices.length),
-                Arrays.copyOf(previousMetadata.visibleOpenIndices, previousMetadata.visibleOpenIndices.length),
-                Arrays.copyOf(previousMetadata.allClosedIndices, previousMetadata.allClosedIndices.length),
-                Arrays.copyOf(previousMetadata.visibleClosedIndices, previousMetadata.visibleClosedIndices.length),
+                previousMetadata.allIndices,
```

Comment:
Why is this also not a copy? Places where we create temporary metadata objects and mutate may lead to issues.

Reply:
Previously these data structures were arrays. Now these are immutable collections(unmodifiable collection). So these copying are not required.